my-app/src/frontend/components/navbar/Navbar.js

```diff
@@ -3,19 +3,47 @@ import { FaShoppingCart } from "react-icons/fa";
 import { AiFillHeart } from "react-icons/ai";
 import "./navbar.css";
 import "../../../App.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../../contexts/auth-context";
 
 function Navbar() {
+  const { authState, authDispatch } = useAuth();
+  const { token } = authState;
+  const navigate = useNavigate();
+
+  function logoutHandler(e) {
+    e.preventDefault();
+    localStorage.removeItem("token");
+    authDispatch({ type: "TOKEN", payload: null });
+    navigate("/");
+  }
+
+  function loginHandler() {
+    navigate("/login");
+  }
+
   return (
     <nav class="navigation-bar">
       <div className="upper-nav">
-        <span className="log-in">Login</span>
-        <span className="sign-up">Sign up</span>
+        <Link to="/signup">
+          <button className="sign-up">Sign up</button>
+        </Link>
+        {token ? (
+          <button className="log-out" onClick={logoutHandler}>
+            Logout
+          </button>
+        ) : (
+          <button className="log-in" onClick={loginHandler}>
+            Login
+          </button>
```

Comment:
Suggestion : use Link here --> `<Link to="/login" className="log-in">Login</Link>